source/extensions/common/wasm/wasm.cc

```diff
@@ -1213,10 +1217,12 @@ std::unique_ptr<WasmVm> createWasmVm(absl::string_view wasm_vm) {
   }
 }
 
-std::unique_ptr<Wasm> createWasm(absl::string_view id,
-                                 const envoy::config::wasm::v2::VmConfig& vm_config,
-                                 Api::Api& api) {
-  auto wasm = std::make_unique<Wasm>(vm_config.vm(), id);
+static Wasm* createWasmInternal(absl::string_view id,
```

Comment:
I don't get the point of `createWasmInternal`
If it is used by creating share_ptr and unique_ptr, the better shape is delegating create shared_ptr to create unique_ptr. See my comment in crateWasm.

Reply:
Removed as part of going to only shared_ptr.